Classes/Utility/Image+Base64.swift

```diff
@@ -0,0 +1,31 @@
+//
+//  Image+Base64.swift
+//  Freetime
+//
+//  Created by Sherlock, James on 18/10/2017.
+//  Copyright © 2017 Ryan Nystrom. All rights reserved.
+//
+
+import UIKit
+
+extension UIImage {
+    
+    /// Compressed and Encodes in Base64 the given image.
+    ///
+    /// Process is moved to a background thread in order to prevent UI blocking.
+    ///
+    /// Compression is a value between 0.0 and 1.0. Lower is smaller file size but worse quality.
+    func compressAndEncode(compression: CGFloat = 0.2, completion: @escaping (Result<String>) -> Void) {
```

Comment:
Hot tip that `0.65` is a great compression v quality ratio. Much experience here 😉

Reply:
We can make the make the number bigger here, by all means - the issue is the Imgur API has a maximum file size. So if the number is too big (I had it fail with a 0.8) then the image will just fail. 0.2 is probably a bit too extreme though 😅